src/log/log.c

```diff
@@ -705,7 +705,7 @@ __log_decompress(WT_SESSION_IMPL *session, WT_ITEM *in, WT_ITEM *out)
 	conn = S2C(session);
 	logrec = (WT_LOG_RECORD *)in->mem;
 	skip = WT_LOG_COMPRESS_SKIP;
-	compressor = conn->log_compressor;
+	compressor = conn->recovery_log_compressor;
```

Comment:
This code should use the correct compressor based on whether or not we're in recovery. Something like a `printlog` would be decompressing log records and should not use the recovery one.